src/users_router.rs

```diff
@@ -86,10 +89,55 @@ impl UsersRouter {
         Ok(Response::with(status::NotImplemented))
     }
 
-    pub fn new() -> iron::middleware::Chain {
+    fn setup(req: &mut Request) -> IronResult<Response> {
+        #[derive(RustcDecodable, Debug)]
+        struct SetupBody {
+            username: String,
+            email: String,
+            password: String
+        }
+
+        let mut payload = String::new();
+        req.body.read_to_string(&mut payload).unwrap();
```

Comment:
Can this cause a panic if the payload is malformed, or binary or is not present?